gcloud-java-storage/src/main/java/com/google/gcloud/storage/Storage.java

```diff
@@ -501,6 +502,14 @@ public static Builder builder() {
   BlobInfo create(BlobInfo blobInfo, byte[] content, BlobTargetOption... options);
 
   /**
+   * Create a new blob.
+   *
+   * @return a complete blob information.
+   * @throws StorageException upon failure
+   */
+  BlobInfo createFromStream(BlobInfo blobInfo, InputStream content, BlobTargetOption... options);
```

Comment:
Honestly, I am not sure why I cared to allow `null` here in the first place. I think that should be discouraged and if we want to support it we can also provide `datastore.create(blobInfo)`.